substrate/primitives/api/Cargo.toml

```diff
@@ -21,13 +21,16 @@ sp-api-proc-macro = { path = "proc-macro", default-features = false }
 sp-core = { path = "../core", default-features = false }
 sp-std = { path = "../std", default-features = false }
 sp-runtime = { path = "../runtime", default-features = false }
+sp-runtime-interface = { path = "../runtime-interface", default-features = false }
```

Comment:
I think we should directly import polkavm here. No need to pull in the entire `sp-runtime-interface`.

Reply:
The problem here is that this needs access to the `polkavm_abi` module we've defined in `sp-runtime-interface`, so just directly including the PolkaVM crates is not sufficient.

So I can think of three options here:
1) Define the `polkavm_abi` module in `sp-runtime-interface` and depend on it here.
2) Define the `polkavm_abi` in `sp-wasm-interface` (which actually isn't WASM specific, since PolkaVM support also uses what's defined in there because it reuses parts of the existing machinery to avoid having to rewrite everything, *and* the `wasmtime` feature for that crate is optional anyway; a better name for that crate would be, uh, `sp-runtime-interface`, but that's already taken, so maybe merging these two crates would be the way to go :P)
3) Define the `polkavm_abi` in a new crate, say, `sp-polkavm-interface`, and depend on it here. (But that seems like an overkill to me to add a new crate for ~10 lines of code, and also because the stuff in `sp-wasm-interface` is also used it'd be weird to have WASM only use `sp-wasm-interface` but PolkaVM use both `sp-polkavm-interface` and `sp-wasm-interface`.)

What I would suggest we do here is:

1) Leave the `polkavm_abi` definition in `sp-runtime-interface` and depend on it here.
2) (In another PR) Remove the `sp-wasm-interface` crate and merge it with `sp-runtime-interface`. Then refactor `sp-runtime-interface` so that it's more modularized and you don't need to always pull in the whole thing.

Does this sound good to you? (Alternatively maybe you have a better idea?)

Reply:
Ahh okay, yeah then fine to leave it the way you have done it right now.